aten/src/ATen/native/ReflectionPad.cpp

```diff
@@ -30,12 +31,12 @@ TORCH_META_FUNC(reflection_pad1d)(const Tensor& input, IntArrayRef padding) {
   int64_t dim_w = 1;
   int64_t nbatch = 1;
 
-  // allow dim=0 only in the batch dimension.
-  TORCH_CHECK(
-      (input.ndimension() == 2 && input.size(1) != 0) ||
-          (input.ndimension() == 3 && input.size(1) != 0 && input.size(2) != 0),
-      "2D or 3D (batch mode) tensor expected for input, but got: ",
-      input);
+  TORCH_CHECK(padding.size() == 2, "padding size is expected to be 2");
+  auto pad_l = padding[0];
+  auto pad_r = padding[1];
+
+  // allow empty batch size but not other dimensions.
+  at::native::padding::check_valid_input<1>(input);
```

Comment:
Thanks for making the updates @mingfeima ! I think it looks much better. I do have to ask for another small refactor so I can more easily accept and defend this change.

While I agree that this unifying function is useful, I have to worry that we're maintaining the same error messages and error behavior across both code paths.

Can you split the refactor of these input sanitization functions into PRs below this stack?

I want to make sure we separate the task of "refactor existing error checking code for both CPU and CUDA" from "add an optimized kernel for a new feature" and from "add a new feature (channels last)". I think this PR currently does those three things at once.

Also the error checking functions should be within this file and not in the cpu subfolder since they're not cpu specific and we don't need to compile them multiple times for various instruction sets.

Thank you!

Reply:
OK, sure will have it done :)